.github/workflows/build-end-to-end-test.yaml

```diff
@@ -54,16 +149,72 @@ jobs:
           name: test-results
           path: github-artifact
 
-      - name: Upload Allure results to GitHub Artifact
-        uses: actions/upload-artifact@v2
+      - name: Check cache
         if: always()
-        with:
-          name: test-results
-          path: allure-reports
+        run: |
+          pwd
+          ls ${{ inputs.environment }}-${{ inputs.test }}-test-histories/*
+
+      - name: Init/Update cache
+        if: always()
+        run: |
+          pwd
+          cp -a -r ${{ inputs.environment }}-${{ inputs.test }}-test-histories/* ${{ inputs.test }}/allure-results
+          echo "----------------------------------->>> Allure-results after merged"
+          ls ${{ inputs.test }}/allure-results/*
+          mkdir ${{ inputs.test }}/allure-report2
+          allure generate ${{ inputs.test }}/allure-results -o ${{ inputs.test }}/allure-report2 --clean
+          echo "----------------------------------->>> New Allure report"
+          ls ${{ inputs.test }}/allure-report2/*
+          cp -a -r ${{ inputs.test }}/allure-report2/history ${{ inputs.environment }}-${{ inputs.test }}-test-histories
+
+      - name: Check cache after generating new report
+        if: always()
```

Comment:
cache is something to be wiped without impact, i do not think we can rely on it being always there and not cleaned say by github itself periodically

as i understand this is used as sort of storage right now, but i do not see it as a valid approach even if it works right now

i think it makes sense to create a followup to implement a persistent solution that does not rely on github cache